spec/features/user_adds_new_review_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+
+feature 'user adds new review', %{
+  As a user
+  I want to review dinosaurs
+  So that I can share my opinions
+
+  Acceptance Criteria
+  [x] User sees the name of the dinosaur and list of reviews
+  [x] User must fill out form correctly
+  [x] User will see review on dino page if correctly filled
+  [x] User will see error if form is incorrect
+
+} do
+  feature "User is not signed in" do
+    scenario "user adds review before sigining in" do
+      dino = FactoryGirl.create(:dinosaur)
+
+      visit dinosaur_path(dino)
+      fill_in("Body", with: "What a nice dinosaur!")
+      fill_in("Rating", with: 2)
+      click_button("Submit")
+
+      expect(page).to have_content("Sign In")
+    end
+  end
+
+  feature "User is signed in" do
+
+    before(:each) do
+      user = FactoryGirl.create(:user)
+      visit new_user_session_path
+      fill_in 'Email', with: user.email
+      fill_in 'Password', with: user.password
+      click_button 'Log in'
+    end
+
+    scenario "visit details page to see name and details" do
+      dino = FactoryGirl.create(:dinosaur)
+      review = FactoryGirl.create(:review, dinosaur_id: dino.id)
```

Comment:
You can combine lines 39 and 40 by just creating a new Factory girl `review` object and then referring to the dino it will make via that object: i.e. `expect(page).to have_content(review.dino.name)`. Just a way to DRY things up and create fewer variables in here.